Mathlib/NumberTheory/NumberField/AdeleRing.lean

```diff
@@ -0,0 +1,151 @@
+/-
+Copyright (c) 2024 Salvatore Mercuri, María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Salvatore Mercuri, María Inés de Frutos-Fernández
+-/
+import Mathlib.NumberTheory.NumberField.CanonicalEmbedding.Basic
+import Mathlib.NumberTheory.NumberField.Completion
+import Mathlib.RingTheory.DedekindDomain.FiniteAdeleRing
+
+/-!
+# The adele ring of a number field
+
+This file contains the formalisation of the infinite adele ring of a number field as the
+finite product of completions over its infinite places and the adele ring of a number field as the
+direct product of the infinite adele ring and the finite adele ring.
+
+## Main definitions
+ - `NumberField.InfiniteAdeleRing` of a number field `K` is defined as the product of
+   the completions of `K` over its infinite places.
+ - `NumberField.InfiniteAdeleRing.ringEquiv_mixedSpace` is the ring isomorphism between
+   the infinite adele ring of `K` and `ℝ ^ r₁ × ℂ ^ r₂`, where `(r₁, r₂)` is the signature of `K`.
+ - `NumberField.AdeleRing K` is the adele ring of a number field `K`.
+ - `NumberField.AdeleRing.principalSubgroup K` is the subgroup of principal adeles `(x)ᵥ`.
+
+## Main results
+ - `NumberField.InfiniteAdeleRing.locallyCompactSpace` : the infinite adele ring is a
+   locally compact space.
+
+## References
+ * [J.W.S. Cassels, A. Frölich, *Algebraic Number Theory*][cassels1967algebraic]
+
+## Tags
+infinite adele ring, adele ring, number field
+-/
+
+noncomputable section
+
+namespace NumberField
+
+open InfinitePlace AbsoluteValue.Completion InfinitePlace.Completion DedekindDomain IsDedekindDomain
+
+open scoped Classical
+
+variable (K : Type*) [Field K] (v : InfinitePlace K)
```

Comment:
```
variable (K : Type*) [Field K]
```
You are only using the variable v at two places so I think it is better to define it there (see also below) than globally.

Reply:
Agree, changes made!